tests/HttpHandlerTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bauhaus\HttpHandler;
+
+use Bauhaus\HttpHandler\Double\MockResponseFactory;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+
+class HttpHandlerTest extends TestCase
+{
+    private ResponseFactoryInterface $factory;
+
+    protected function setUp(): void
+    {
+        $this->factory = new MockResponseFactory($this->createMock(ResponseInterface::class));
+    }
+
+    public function testWhenRouteIsNotFoundThenThrowException(): void
+    {
+        $uri = $this->createMock(UriInterface::class);
+        $uri->method('getPath')->willReturn('/');
+        $request = $this->createMock(ServerRequestInterface::class);
+        $request->method('getMethod')->willReturn('GET');
+        $request->method('getUri')->willReturn($uri);
```

Comment:
I would just extract the creation of a request to a method `$this->request('GET /')`.